3-UsingAPIs/Week2/homework/ex3-rollAnAce.js

```diff
@@ -12,23 +12,22 @@ Full description at: https://github.com/HackYourFuture/Homework/blob/main/3-Usin
 // ! Do not change or remove the next two lines
 const rollDie = require('../../helpers/pokerDiceRoller');
 
-function rollDieUntil(wantedValue) {
-  // TODO: rewrite this function using async/await
-  return rollDie().then((value) => {
-    if (value !== wantedValue) {
-      return rollDieUntil(wantedValue);
-    }
-    return value;
-  });
+async function rollDieUntil(wantedValue) {
+  while (true) {
+    const value = await rollDie();
+    if (value !== wantedValue) continue;
+    else return value;
+  }
```

Comment:
This is what I would have expected: use the condition of `while` loop rather than an embedded `if` statement.

```suggestion
  let value;   // -> initially undefined
  while (value !== wantedValue) {
    value = await rollDie();
  }
  
  return value;
```